### PR TITLE
Convert repo to RN TypeScript template

### PR DESCRIPTION
**Pull request by luisfmsouza**
Convert repo to RN TypeScript template.

- Upgrade README instructions

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -1,20 +1,36 @@
...
-      <Text>Welcome to the FindHotel mobile app coding assignment!</Text>
-      <StatusBar style="auto" />
-    </View>
-  );
-}
+const App = () => (
+  <View style={styles.sectionContainer}>
+    <Text style={styles.title}>Welcome!</Text>
+    <Text style={styles.description}>
+      This is the repository for <Text style={styles.highlight}>FindHotel</Text>{' '}
```
Comment (jopdeklein) [inline App.tsx]:
```suggestion
      This is the repository for the <Text style={styles.highlight}>FindHotel</Text>{' '}
```

File: README.md [hunk 2]
```diff
@@ -1,26 +1,24 @@
...
+## How to run
 
-This project was bootstrapped using Expo. In the project directory, you can run:
-
-_Note: you will first need to install the [Expo CLI](https://docs.expo.dev/workflow/expo-cli/) globally._
+This project was bootstrapped using [React Native TypeScript template](https://reactnative.dev/docs/typescript). In the project directory, you can run:
 
 ### `npm install`
 
 To have all the dependencies installed.
```
Comment (jopdeklein) [inline README.md]: 🐜  Not related to your changes, but: 

```suggestion
Installs all the dependencies.
```

File: ios/Podfile [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+target 'FHApp' do
+  config = use_native_modules!
+
+  # Flags change depending on the env values.
+  flags = get_default_flags()
+
+  use_react_native!(
+    :path => config[:reactNativePath],
+    # to enable hermes on iOS, change `false` to `true` and then install pods
```
Comment (jopdeklein) [inline ios/Podfile]: Not sure if this was modified or an issue with the scaffolding, but I am missing the `false`  to change? Probably not important but not sure if candidates want to change it for whatever reason.

**Review by jopdeklein — APPROVED** [review]
I hope you don't mind I didn't review the generated files in depth, but I was able to run the demo app :)

File: ios/Podfile (hunk 3 above)
Comment (luisfmsouza) [reply ios/Podfile]: In the latest version, we don't have the boolean hardcoded anymore (you can check [the docs here](https://reactnative.dev/docs/next/hermes#ios)). I've seen a open pull request to clarify this doc, but I'm not sure if is already released or not.

We can also [see here](https://github.com/react-native-community/upgrade-support/issues/193#issuecomment-1158371500) about how to turn it on or off.

I'll document it better on our repo.

Comment (luisfmsouza) [reply ios/Podfile]: [d0e496a](https://github.com/FindHotel/fh-app-assignment/pull/6/commits/d0e496a4cec07c268da2ee7f1168e67c0f6120db)

**Pull request merged**